src/Json.Schema.ToDotNet/ComparerCodeGenerator.cs

```diff
@@ -0,0 +1,1603 @@
+                            .AddArgumentListArguments(
+                                SyntaxFactory.Argument(
+                                    SyntaxFactory.IdentifierName(FirstEqualsArgumentName)),
+                                SyntaxFactory.Argument(
+                                    SyntaxFactory.IdentifierName(SecondEqualsArgumentName)))))));
+
+            return methods;
+        }
+
+        #endregion
+
+        private StatementSyntax DeclareCompareResultLocalVariable(string variableName)
+        {
+            /*
+            int compareResult = 0;
+            */
+            return SyntaxFactory.LocalDeclarationStatement(
+                SyntaxFactory.VariableDeclaration(
+                    SyntaxFactory.PredefinedType(
+                        SyntaxFactory.Token(SyntaxKind.IntKeyword)))
+                .AddVariables(
+                    SyntaxFactory.VariableDeclarator(
+                        SyntaxFactory.Identifier(variableName))
+                    .WithInitializer(
+                        SyntaxFactory.EqualsValueClause(
+                            SyntaxFactory.LiteralExpression(
+                                SyntaxKind.NumericLiteralExpression,
+                                SyntaxFactory.Literal(0))))));
+        }
+
+        private StatementSyntax AugumentNullCheck(string argumentName)
+        {
+            /*
+            if (argument == null)
+            {
+                throw new ArgumentNullException(nameof(argument));
+            }
+            */
+            return SyntaxFactory.IfStatement(
+                SyntaxHelper.IsNull(argumentName),
+                SyntaxFactory.Block(
+                    SyntaxFactory.SingletonList<StatementSyntax>(
+                        SyntaxFactory.ThrowStatement(
+                            SyntaxFactory.ObjectCreationExpression(
+                                SyntaxFactory.IdentifierName("ArgumentNullException"))
+                            .AddArgumentListArguments(
+                                SyntaxFactory.Argument(
+                                    SyntaxFactory.InvocationExpression(
+                                        SyntaxFactory.IdentifierName(
+                                            SyntaxFactory.Identifier(
+                                                SyntaxFactory.TriviaList(),
+                                                SyntaxKind.NameOfKeyword,
+                                                "nameof",
+                                                "nameof",
+                                                SyntaxFactory.TriviaList())))
+                                    .AddArgumentListArguments(
+                                            SyntaxFactory.Argument(
+                                                SyntaxFactory.IdentifierName(argumentName)))))))));
+        }
+
+        private StatementSyntax CallTryReferenceComparesMethod()
```

Comment:
>CallTryReferenceComparesMethod [](http://example.com/codeflow?start=32&length=30)

Update this helper to add this comment before the call:\

            // TryReferenceCompares is an autogenerated extension method
            // that will properly handle the case when 'left' is null.